js/src/views/Home/Urls/urls.js

```diff
@@ -0,0 +1,139 @@
+// along with Parity.  If not, see <http://www.gnu.org/licenses/>.
+
+import { observer } from 'mobx-react';
+import moment from 'moment';
+import React, { Component, PropTypes } from 'react';
+import { FormattedMessage } from 'react-intl';
+
+import { Container, ContainerTitle, DappUrlInput, SectionList } from '~/ui';
+import { LinkIcon } from '~/ui/Icons';
+
+import styles from './urls.css';
+
+@observer
+export default class Urls extends Component {
+  static contextTypes = {
+    router: PropTypes.object.isRequired
+  };
+
+  static propTypes = {
+    extensionStore: PropTypes.object.isRequired,
+    store: PropTypes.object.isRequired
+  }
+
+  render () {
+    const { nextUrl } = this.props.store;
+
+    return (
+      <div className={ styles.urls }>
+        <div className={ styles.layout }>
+          <ContainerTitle
+            title={
+              <FormattedMessage
+                id='home.url.title'
+                defaultMessage='Web Applications'
+              />
+            }
+          />
+          <DappUrlInput
+            className={ styles.input }
+            onChange={ this.onChangeUrl }
+            onGoto={ this.onGotoUrl }
+            onRestore={ this.onRestoreUrl }
+            url={ nextUrl }
+          />
+          { this.renderHistory() }
+        </div>
+      </div>
+    );
+  }
+
+  renderHistory () {
+    const { history } = this.props.store;
+
+    if (!history.length) {
+      return (
+        <div className={ styles.empty }>
+          <FormattedMessage
+            id='home.url.none'
+            defaultMessage='No recent URL history available'
+          />
+        </div>
+      );
+    }
+
+    return (
+      <SectionList
+        items={ history }
+        renderItem={ this.renderHistoryItem }
+      />
+    );
+  }
+
+  renderHistoryItem = (history) => {
+    if (!history || !history.url) {
+      return null;
+    }
+
+    const onNavigate = () => this.onGotoUrl(history.url);
+
+    return (
+      <Container
+        className={ styles.historyItem }
+        hover={
+          <div className={ styles.timestamp }>
+            <FormattedMessage
+              id='home.url.visited'
+              defaultMessage='visited {when}'
+              values={ {
+                when: moment(history.timestamp).fromNow()
+              } }
+            />
+          </div>
+        }
+        key={ history.timestamp }
+        onClick={ onNavigate }
+      >
+        <LinkIcon className={ styles.linkIcon } />
+        <div className={ styles.url }>
+          { history.hostname }
+        </div>
+      </Container>
+    );
+  }
+
+  onChangeUrl = (url) => {
+    this.props.store.setNextUrl(url);
+  }
+
+  onGotoUrl = (url) => {
+    const { router } = this.context;
+    const { extensionStore } = this.props;
+
+    this.props.store.gotoUrl(url);
+
+    if (extensionStore.hasExtension) {
+      window.open(this.props.store.currentUrl, '_blank');
+    } else {
+      router.push('/web');
```

Comment:
Yes ok, I haven't seen that the Browser dapp don't take into account the *real* browser url.